int8 mode
float64 virtual_center_x
float64 virtual_center_y
float64 direction
